pkg/scheduler/frameworkext/scheduler_adapter.go

```diff
@@ -69,6 +72,58 @@ func (s *SchedulerAdapter) MoveAllToActiveOrBackoffQueue(event framework.Cluster
 	s.Scheduler.SchedulingQueue.MoveAllToActiveOrBackoffQueue(event, nil)
 }
 
+var _ SchedulerCache = &cacheAdapter{}
+
+type cacheAdapter struct {
+	scheduler *scheduler.Scheduler
+}
+
+func (c *cacheAdapter) AddPod(pod *corev1.Pod) error {
+	return c.scheduler.Cache.AddPod(pod)
+}
+
+func (c *cacheAdapter) UpdatePod(oldPod, newPod *corev1.Pod) error {
+	return c.scheduler.Cache.UpdatePod(oldPod, newPod)
+}
+func (c *cacheAdapter) RemovePod(pod *corev1.Pod) error {
+	return c.scheduler.Cache.RemovePod(pod)
+
+}
+func (c *cacheAdapter) AssumePod(pod *corev1.Pod) error {
+	return c.scheduler.Cache.AssumePod(pod)
+}
+
+func (c *cacheAdapter) IsAssumedPod(pod *corev1.Pod) (bool, error) {
+	return c.scheduler.Cache.IsAssumedPod(pod)
+}
+
+func (c *cacheAdapter) GetPod(pod *corev1.Pod) (*corev1.Pod, error) {
+	return c.scheduler.Cache.GetPod(pod)
+}
+
+func (c *cacheAdapter) ForgetPod(pod *corev1.Pod) error {
+	return c.scheduler.Cache.ForgetPod(pod)
+}
+
+func (c *cacheAdapter) InvalidNodeInfo(nodeName string) error {
```

Comment:
```
// SetNode sets the overall node information.
func (n *NodeInfo) SetNode(node *v1.Node) {
	n.node = node
	n.Allocatable = NewResource(node.Status.Allocatable)
	n.Generation = nextGeneration()
}
```

We can also use `SetNode` to update generation, maybe this cost is lower?

Reply:
Yes, using `SetNode` is also a good method, but this may conflict with the Node's Update Event, because the version of Node we get when we call it may be later than the latest version.

Reply:
That make sense